internal/component/mimir/rules/kubernetes/events.go

```diff
@@ -202,6 +219,62 @@ func (e *eventProcessor) desiredStateFromKubernetes() (kubernetes.RuleGroupsByNa
 	return desiredState, nil
 }
 
+func addMatchersToQuery(query string, matchers []Matcher) (string, error) {
+	var err error
+	for _, s := range matchers {
+		query, err = labelsSetPromQL(query, s.MatchType, s.Name, s.Value)
+		if err != nil {
+			return "", err
+		}
+	}
+	return query, nil
+}
+
+// Lifted from: https://github.com/prometheus/prometheus/blob/79a6238e195ecc1c20937036c1e3b4e3bdaddc49/cmd/promtool/main.go#L1242
+func labelsSetPromQL(query, labelMatchType, name, value string) (string, error) {
+	expr, err := parser.ParseExpr(query)
+	if err != nil {
+		return query, err
+	}
+
+	var matchType promlabels.MatchType
+	switch labelMatchType {
+	case parser.ItemType(parser.EQL).String():
+		matchType = promlabels.MatchEqual
+	case parser.ItemType(parser.NEQ).String():
+		matchType = promlabels.MatchNotEqual
+	case parser.ItemType(parser.EQL_REGEX).String():
+		matchType = promlabels.MatchRegexp
+	case parser.ItemType(parser.NEQ_REGEX).String():
+		matchType = promlabels.MatchNotRegexp
+	default:
+		return query, fmt.Errorf("invalid label match type: %s", labelMatchType)
+	}
```

Comment:
this could be checked at the Validate step to exit early if a match block is not configured with a valid match type

Reply:
It is being checked in the Validate step: https://github.com/grafana/alloy/pull/1773/files/453e253075f1a450a26240b8e6a94f4590e49c60#diff-1103eec8a21dbf8e36fdaf679f6214c70a74407b70d7786e3e826cdeab48f1e4R90 

This code is copied from upstream promtool (there's no way to import it as it's in main.go). So that check is redundant and comes from there. I'd prefer to leave this as-is.

Reply:
Also, see this test case that uses Validate: https://github.com/grafana/alloy/pull/1773/files/453e253075f1a450a26240b8e6a94f4590e49c60#diff-5fe9918a1cab895fbb9ac0ec979e437c157885dce549f3920b0a6f4f40e08392R82

Reply:
aa my bad I missed it, thanks